src/Interpreters/Aggregator.cpp

```diff
@@ -282,6 +286,26 @@ DB::ColumnNumbers calculateKeysPositions(const DB::Block & header, const DB::Agg
         keys_positions[i] = header.getPositionByName(params.keys[i]);
     return keys_positions;
 }
+
+template <typename HashTable, typename KeyHolder>
+concept HasPrefetchMemberFunc = requires
+{
+    {std::declval<HashTable>().prefetch(std::declval<KeyHolder>())};
+};
+
+size_t getMinBytesForPrefetch()
+{
+    size_t l2_size = 0;
+
+#if defined(OS_LINUX) && defined(_SC_LEVEL2_CACHE_SIZE)
+    if (auto ret = sysconf(_SC_LEVEL2_CACHE_SIZE); ret != -1)
+        l2_size = ret;
+#endif
+
+    /// 1MB looks like a reasonable estimate of L2 size. 4 is empirical constant.
+    return 4 * std::max<size_t>(l2_size, 1024 * 1024);
```

Comment:
Why std::max?

PS. L2 cache is typically 256 KiB per core on Intel CPU.
Btw, what sysconf returns? The per-core size or total size?

Reply:
On my CPU it is 512 KiB/core, 32 MB in total.

Reply:
Let's replace 1024 * 1024 to 256 * 1024.

Reply:
> Why std::max?

if HT fits in l2, prefetching will likely harm perf, so the threshold should be greater than l2 size.

> Btw, what sysconf returns? The per-core size or total size?

per-core

on my vm it is 1MB. my point was that it is safer to choose greater value than smaller. but we could use 256KB, arrow use it as default for l2.